pkg/volume/cephfs/cephfs.go

```diff
@@ -301,13 +309,29 @@ func (cephfsVolume *cephfs) execMount(mountpoint string) error {
 	return nil
 }
 
-func getVolumeSource(spec *volume.Spec) (*v1.CephFSVolumeSource, bool, error) {
+func getVolumeSource(spec *volume.Spec) (mon []string, path, user, secretFile string, localSecretRef *v1.LocalObjectReference, secretRef *v1.ObjectReference, readOnly bool, err error) {
```

Comment:
`localSecretRef *v1.LocalObjectReference, secretRef *v1.ObjectReference` doesn't seem great.

I'd rather see something like this:
```
func getSecretReference(podNamespace *string, spec *volume.Spec) (SecretReference, error)
```

that is responsible for determining the full secret reference, name and namespace, and returns an error in the following cases:
* podNamespace is not provided, and the volume is a local volume (spec.Volume.CephFS)
* podNamespace is not provided, and the volume is a PV volume and has no namespace specified